ddtrace/_trace/tracer.py

```diff
@@ -406,27 +407,34 @@ def current_trace_context(self, *args, **kwargs) -> Optional[Context]:
             return active.context
         return None
 
-    def get_log_correlation_context(self) -> Dict[str, str]:
+    def get_log_correlation_context(self, active: Optional[Union[Context, Span]] = None) -> Dict[str, str]:
         """Retrieves the data used to correlate a log with the current active trace.
         Generates a dictionary for custom logging instrumentation including the trace id and
         span id of the current active span, as well as the configured service, version, and environment names.
         If there is no active span, a dictionary with an empty string for each value will be returned.
         """
-        active: Optional[Union[Context, Span]] = None
-        if self.enabled or self._apm_opt_out:
+        if active is None and (self.enabled or self._apm_opt_out):
```

Comment:
What's the difference between _apm_opt_out and enabled? 
```
        if active is None and (self.enabled or not self._apm_opt_out):
```

Reply:
this was already there so i thought it may be important to keep

Reply:
That's fine we should probably fix this in a follow up PR. I am pretty confident it's a bug. It doesn't make sense to do log correlation when `apm_opt_out = True` and NOT do log correlation when `apm_opt_out = False`. 

Also `apm_opt_out` does not prevent spans from being sent to the agent. I believe the agent receives this flag as a span tag and then does something (maybe drops the spans 🤷). In the past we always injected span_ids and trace_ids into logs if tracing was enabled. We never took into account the span's sampling decision. I am not sure why we even need to check `apm_opt_out`.

Reply:
sounds good